examples/erc20/aETH/aETH.xml

```diff
@@ -0,0 +1,403 @@
+                <ts:string xml:lang="en">Borrow DAI</ts:string>
+            </ts:label>
+            <ts:attribute name="borrowAmount">
+                <ts:type><ts:syntax>1.3.6.1.4.1.1466.115.121.1.36</ts:syntax></ts:type>
+                <ts:label>
+                    <ts:string xml:lang="en">Amount to Borrow</ts:string>
+                </ts:label>
+                <ts:origins>
+                    <!-- e18 is a hard coded multiplier.
+                    rationale for hardcoding: avoiding over-design  -->
+                    <ts:user-entry as="e18"/>
+                </ts:origins>
+            </ts:attribute>
+            <!-- 1 is stable rate, 2 is variable rate -->
+            <ts:attribute name="interestRateModel">
+                <ts:type><ts:syntax>1.3.6.1.4.1.1466.115.121.1.36</ts:syntax></ts:type>
+                <ts:label>
+                    <ts:string xml:lang="en">choice between variable and fixed interest rate</ts:string>
+                </ts:label>
+                <ts:origins>
+                    <!-- e18 is a hard coded multiplier.
+                    rationale for hardcoding: avoiding over-design  -->
+                    <ts:user-entry as="uint"/>
+                </ts:origins>
+            </ts:attribute>
+            <ts:transaction>
+                <ethereum:transaction function="borrow" contract="LendingPoolProvider" as="uint">
+                    <ts:data>
+                        <ts:address>0x6b175474e89094c44da98b954eedeac495271d0f</ts:address>
+                        <ts:uint256 ref="borrowAmount"/>
+                        <ts:uint256 ref="interestRateModel"/>
+                        <ts:uint256>0</ts:uint256> <!-- referral code -->
+                    </ts:data>
+                </ethereum:transaction>
+            </ts:transaction>
+            <ts:view xml:lang="en" xmlns="http://www.w3.org/1999/xhtml">
+                <xhtml:style type="text/css">&style;</xhtml:style>
+                <xhtml:script  type="text/javascript">&borrow.en;</xhtml:script>
+            </ts:view>
+        </ts:card>
+
+
+        <ts:card type="activity" name="aETHRedeemed">
+            <ts:origins>
+                <ethereum:event type="Redeem" filter="ownerAddress=${_fromRedeemed}"/>
+            </ts:origins>
+            <ts:item-view xml:lang="en" xmlns="http://www.w3.org/1999/xhtml">
+                <style type="text/css">&style;</style>
+                <xhtml:script  type="text/javascript">&item-view-redeemed.en;</xhtml:script>
+            </ts:item-view>
+            <ts:view xml:lang="en" xmlns="http://www.w3.org/1999/xhtml">
+                <xhtml:style type="text/css">&style;</xhtml:style>
+                <xhtml:script  type="text/javascript">&item-view-redeemed.en;</xhtml:script>
+            </ts:view>
+        </ts:card>
+
+        <ts:card type="activity" name="aETHBorrowed">
+            <ts:origins>
+                <ethereum:event type="Borrow" filter="ownerAddress=${_user}"/>
```

Comment:
There isn't a matching module type defined for this.

Reply:
`filter="ownerAddress=${_user}"`

The order the filter is written seems a bit strange to me. I thought this would be a bit more natural:

`filter="_user=${ownerAddress}"`

As in:

`filter="attribute=value"` rather than `filter="value=attribute"`

But I *think* it works syntactically (will have to test), so just an observation.

Reply:
I think the interest rate might be too good to be true :)

Reply:
@hboon changed, will check up on the interest rate later